### PR TITLE
Feature/#22 lightsaber

### LINKED ISSUE #22: UI Component | Lightsaber displaying remaining time
old hash: #16

### PR DESCRIPTION
**Pull request by SebastianBabinski1**
closes #22 
Height/width values come from figma project.

### CODE REVIEW

File: src/app/components/Lightsaber.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+function Lightsaber() {
+  const root = document.querySelector('.remaining-time');
+  const divElement = document.createElement('div');
+  const handle = document.createElement('div');
+  const blade = document.createElement('div');
+  const redBlade = document.createElement('div');
+
+  divElement.classList.add('remaining-time__lightsaber');
+  divElement.classList.add('lightsaber');
```
Comment (Enessetere) [inline src/app/components/Lightsaber.js:9]: While adding classes to one element, separate classes with ','

`divElement.classList.add('remaining-time__lightsaber', 'lightsaber');`

**Review by Enessetere — CHANGES_REQUESTED** [review]
Looks good to me. Please update to develop and resolve conflicts

File: src/app/components/Lightsaber.js (hunk 1 above)
Comment (DominikNowak) [reply src/app/components/Lightsaber.js:9]: Also consider refactor with elementFactory

File: index.html [hunk 2]
```diff
@@ -16,7 +16,8 @@
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
```
Comment (DominikNowak) [inline index.html:16]: This file shouldn't be pushed

**Review by DominikNowak — CHANGES_REQUESTED** [review]

File: src/app/components/RemainingTime.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  parentElement.classList.add('remaining-time');
+  counter.classList.add('remaining-time__counter');
+  counter.textContent = 'Time Left : 02 m 00 s';
+
+  parentElement.appendChild(counter);
+  root.appendChild(parentElement);
+
+  Lightsaber();
+  handleTimeChange(counter);
+};
```
[the remark this answers is not in the document]
Comment (SebastianBabinski1) [reply src/app/components/RemainingTime.js:47]: I did it in PR to #21 and I will use elementFactory during refactorisation

**Comment by SebastianBabinski1** [comment]
I will use elementFactory later, at the same time with remainingTime

**Review by DominikNowak — APPROVED** [review]
Need a refactor, but time is running out, so approved

File: src/app/components/RemainingTime.js [hunk 4]
```diff
@@ -1,9 +1,12 @@
+import { handleBladeSizeChange, Lightsaber } from './Lightsaber';
+
 function handleTimeChange(element) {
-  let timer = process.env.QUIZ_MAX_TIME_SECONDS;
+  const initialTime = process.env.QUIZ_MAX_TIME_SECONDS;
+  let timer = initialTime;
 
   const idInterval = setInterval(() => {
     timer--;
```
Comment (vieraboschkova) [inline src/app/components/RemainingTime.js:8]: Eslint would prefer:
timer -= 1;

File: src/app/components/RemainingTime.js [hunk 5]
```diff
@@ -22,6 +25,8 @@ function handleTimeChange(element) {
     }
 
     element.textContent = `Time Left: ${minToDisplay}m ${secToDisplay}s`;
```
Comment (vieraboschkova) [inline src/app/components/RemainingTime.js:27]: You change the value passed as an argument. Could there be another way to do it?

File: styles/components/_lightsaber.scss [hunk 6]
```diff
@@ -0,0 +1,36 @@
+.lightsaber {
+  align-items: center;
+  display: flex;
+  flex-basis: 100%;
+  order: -1;
```
Comment (vieraboschkova) [inline styles/components/_lightsaber.scss:5]: Please order scss files according to these:
// Positioning
// Display & Box Model
// Color
// Text
// Other

**Review by vieraboschkova — CHANGES_REQUESTED** [review]
Please order scss.

**Review by Enessetere — APPROVED** [review]
Looks good to me, well done.

File: src/app/components/RemainingTime.js [hunk 7]
```diff
@@ -1,9 +1,12 @@
...
+  const initialTime = process.env.QUIZ_MAX_TIME_SECONDS;
+  let timer = initialTime;
 
   const idInterval = setInterval(() => {
     timer--;
-
+    const percentOfTime = ((100 * timer) / initialTime).toFixed(2);
     const min = Math.floor(timer / 60);
     const sec = timer % 60;
     let minToDisplay = min;
```
Comment (Rednaxela700) [inline src/app/components/RemainingTime.js:12]: is `minToDisplay` necessary?

File: styles/components/_lightsaber.scss [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+  &__handle {
+    height: 50px;
+    width: 300px;
+    background: url('../static/assets/ui/LightsaberHandle.png')
+      no-repeat center;
+    transform: translateX(7%);
```
Comment (Rednaxela700) [inline styles/components/_lightsaber.scss:12]: are you sure it's 7%?

File: src/app/components/RemainingTime.js (hunk 5 above)
Comment (SebastianBabinski1) [reply src/app/components/RemainingTime.js:27]: An argument is a div, which contains changing text. We can put there <p> or <span> with text content, but does it change anything?

File: styles/components/_lightsaber.scss (hunk 8 above)
Comment (Rednaxela700) [reply styles/components/_lightsaber.scss:12]: I find it unsafe, as it's parent doesn't have exact width set

**Review by Rednaxela700 — CHANGES_REQUESTED** [review]
1. Update branch merging current develop version
2. Minor fixes
3. Push
4. Let me know

File: src/app/components/RemainingTime.js (hunk 5 above)
Comment (SebastianBabinski1) [reply src/app/components/RemainingTime.js:27]: I declared a variable `const parameter = element`, and used this here. Is it good practice?

File: styles/components/_lightsaber.scss (hunk 8 above)
Comment (SebastianBabinski1) [reply styles/components/_lightsaber.scss:12]: Yes, I have to think about changing this to something safer, but right now I want to implement this to next task. Can I fix it later?

**Review by Rednaxela700 — APPROVED** [review]
Push fixes as we talked on discord and proceed.

**Review by vieraboschkova — APPROVED** [review]

**Pull request merged**